docs/docs/examples/guides/04-object-detection-example.md

```diff
@@ -0,0 +1,745 @@
+converts bounding boxes into a 3D pose. The example works with either an Intel Realsense camera using
+`collections/intel-realsense-collection` or a video played through a custom component which is provided.
+
+## Setup
+
+In AICA Launcher, create a configuration with the following core version and packages:
+
+- AICA Core v4.3.2
+- `collections/object-detection` (TODO: specify version)
+- `collections/intel-realsense-collection v1.0.0`
+
+Under Advanced Settings, add a volume containing the folder where you will store your YOLO model files and link it to `/files`.
+
+## Using the YOLO Executor
+
+The *YOLO Executor* component observes an image from a camera or video feed, runs the YOLO segmentation model on it, and outputs the segmented image as well as the locations of bounding boxes on the image. The predicate `Object Detected` is set to `True` when the object specified by the parameter `Object Class` is found in the image. To test the YOLO
+executor:
+
+- Remove the `*Hardware Interface*`.
+- Add the following components:
+
+  - `Object Detection Components/YOLO Executor`
+  - `Realsense2 Camera/Realsense Camera` if you have an Intel Realsense camera,
+    otherwise, [create a video player component](#create-a-video-player-component) which can be used as a drop in
+    replacement
+
+- Connect both components’ _load nodes_ to load on program start
+- Connect the `RGB Image` output of the `Realsense2 Camera` or `Video Player` to the `RGB Image` input of the *YOLO Executor*
+
+To configure the *YOLO Executor*, set the following parameters:
+
+- Turn on the **auto-configure** and **auto-activate** switches
+- Set `Model Path` to the `.onnx` file, e.g., `/files/yolo12n.onnx`
+- Set `Classes Path` to the yaml label file, e.g., `/files/coco.yaml`
+- Set `Rate` to 3 (works on most machines)
+
+The parameter `Confidence Threshold` is the minimum score a predicted bounding box must have to be considered a valid detection. `IOU Threshold` is used during Non-Maximum Suppression (NMS) to decide whether two bounding boxes represent the same object. For example, if `IOU threshold` is set to 0.5, any box that overlaps more than 50% with a higher-scoring box will be discarded. For now we leave them as default values.
+
+:::info
+
+**Converting YOLO Model files to ONNX format**
+
+The *YOLO Executor* component expects an ONNX file. These can be created from a pretrained [
+`.pt`](https://github.com/sunsmarterjie/yolov12) model (we use the lightweight version — YOLO12n).
+
+Run the following Python script (with [`ultralytics`](https://pypi.org/project/ultralytics/) installed) to convert a
+`.pt` model to `.onnx`:
+
+```python
+from ultralytics import YOLO
+model = YOLO("yolov12n.pt")
+model.export(format="onnx")  # creates 'yolov12n.onnx'
+```
+
+:::
+
+These YOLO models are pre-trained on coco - classes file for
+found [here](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/datasets/coco.yaml). A fine-tuned
```

Comment:
we should probably pick a tag from that repo and not reference main. Otherwise something might change or the path might break before we realize